docs/upgrade/canary-upgrade.md

```diff
@@ -0,0 +1,371 @@
+
+This canary upgrade model is much safer than the in-place upgrade workflow, mainly because it can be done gradually, at your own pace.
+If issues arise they will probably only affect a small portion of your applications, initially, and, if that does happen, it should be much easier to rollback only a part of your application for use with the original control plane.
+
+This upgrade flow gives us more flexibility and confidence when making Istio upgrades, dramatically reducing the potential for service disruptions.
+
+### Canary upgrade process with the Istio operator
+
+Let's take a closer look at how the process works in conjunction with our open source Istio operator.
+So our starting point is a running Istio operator and Istio 1.6 control plane, with applications using that control plane version.
+
+1. Deploy another Istio operator (alongside the previous one), which has Istio 1.7 support
+1. Apply a new Istio CR with Istio version 1.7 and let the operator reconcile an Istio 1.7 control plane
+
+   It is recommended that you turn off gateways for the new control plane and migrate the 1.6 gateway pods to the 1.7 control plane. That way the ingress gateway's typically LoadBalancer-type service will not be recreated, neither will the cloud platform-managed LoadBalancer, and hence the IP address won't change either.
+
+   > This new control plane is often referred to as a *canary*, but, in practice, it is advisable that this be named based on Istio version, as it will remain on the cluster for the long term.
+
+1. Migrate gateways to use the new Istio 1.7 control plane
+
+   This has been intentionally left as a manual step during the new control plane installation with the Istio operator, so that users can make sure that they're ready to use the control plane for their gateways, and only migrate after.
+
+1. Migrate data plane applications
+
+   It is recommended that you perform this migration gradually.
+   It's important to keep in mind during the data plane migration that the two Istio control planes share trust (because they use the same certificates for encrypted communication). Therefore, when an application pod that still uses Istio's 1.6 control plane calls another pod that already uses Istio's 1.7 control plane, the encrypted communication will succeed because of that shared trust.
+   That's why this migration can be performed on a namespace by namespace basis and the communication of the pods won't be affected.
+
+1. Delete the Istio 1.6 control plane
+
+   Once the migration is finished, and we've made sure that our applications are working properly in conjunction with the new control plane, the older Istio 1.6 control plane can be safely deleted.
+   It's recommended that you take some time to make sure that everything is working on the new control plane before turning off the old one; it's only an `istiod` deployment running with practically no data plane pods attached.
```

Comment:
```suggestion
   It's recommended that you take some time to make sure that everything is working on the new control plane before turning off the old one. The overhead of doing so is minimal as it's only an `istiod` deployment running with practically no data plane pods attached.
```

Reply:
I would remove this as it has no real meaning, and I feel like you don't want to go into details here ;-):

practically no data plane pods attached.